### ISSUE TITLE
PNG exported with wrong alpha at vector masks borders

### ISSUE THREAD

**Issue by davidetan** [photopea/photopea#6336]
In my PSD I have some layers using vector masks.
When I export the layers as PNG with Photoshop and Photopea the images are different.
The difference occurs on the vector masks borders.

I'll share a 2x2 image to repreoduce the problem as example: 
[bug.psd.zip](https://github.com/photopea/photopea/files/13978373/bug.psd.zip)

The image has a red layer and a vector mask applied:
<img width="296" alt="image" src="https://github.com/photopea/photopea/assets/7047501/2c967571-5b0a-4ffa-ad6d-8227c5f15594">

The difference is also easy to be seen directly into the two softwares comparing screenshots (screen1 Photopea, screen2 Photoshop):
<img width="147" alt="image" src="https://github.com/photopea/photopea/assets/7047501/0140ac07-a709-4e15-83aa-5620328a1722">
<img width="147" alt="image" src="https://github.com/photopea/photopea/assets/7047501/bc4ba280-cac7-4bf2-a575-4e261024e470">

Here you can find the two exported PNGs: 
![photopea-export](https://github.com/photopea/photopea/assets/7047501/3bd4ad67-ceea-423b-a08b-83f5baeec070)
![photoshop-export](https://github.com/photopea/photopea/assets/7047501/cf22a228-2d00-47ff-b23e-1fa1e4213a2a)

If you open the images using Photopea, you can easily see with the Info tool that the alpha of the top-right pixel is 112 for Photopea and 127 for Photoshop:
<img width="189" alt="image" src="https://github.com/photopea/photopea/assets/7047501/cc0f2ca8-3491-4c8a-b6ce-3717c08046d6">
<img width="189" alt="image" src="https://github.com/photopea/photopea/assets/7047501/0f6b9354-03e7-4d52-acd0-62133ba60734">

Photoshop does not have alpha in the info tool, but putting a black background behind the layer "transfer" the alpha to the red channel and you can see that the value is 127 also into photoshop:
<img width="442" alt="image" src="https://github.com/photopea/photopea/assets/7047501/f56b6ab5-bb88-4b8e-9a83-e715caed0d8a">

-------

I parsed the vector mask pixels in the PSD (channel -2) and it seems that using the same vector, Photoshop and Photopea stores different values.
When you save the PSD I uploaded here with Photoshop and Photopea, the pixel data for channel -2 are changed.

----

Your Operating System: MacOS 14.1.1 (23B81)
Your Web Browser: Google Chrome Version 120.0.6099.216 (Official Build) (x86_64)
The problem happens in the Incognito Mode, too: Yes

**Comment by photopea** [comment]
So it is about anti-aliasing the contours of vector masks. I have changed my code a bit, so it sets the pixel value to 128. 

I also filed a bug report in Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1519670, as we use their engine to rasterize vector graphics.

But I don't know if it is that important. It is only about th contours of vector shapes, and the error was only 5.8% from what it should be: (127-112)/256

**Comment by davidetan** [comment]
Thank you for your support.
Unfortunately the problem still occurs on other images.

I used another image - bigger this time - exported it with Photopea and Photoshop and compared.
This image shows in red the differences on the border ([here the website](https://www.img2go.com/result#j=836dccce-38de-45f2-ad8e-58ebc6c113cf) I used to compare the images)
<img width="238" alt="image" src="https://github.com/photopea/photopea/assets/7047501/97566e20-5471-4c7f-b559-5b2e583435ff">

**Comment by photopea** [comment]
So you are not using the straight lines now, but you use curves? What is the actual difference (in percent)? Why do you think it is "correct" in Photoshop and not in Photopea? :D

**Comment by davidetan** [comment]
> So you are not using the straight lines now, but you use curves?
It's still straight lines, but now there are two vectors.
<img width="735" alt="image" src="https://github.com/photopea/photopea/assets/7047501/649ad323-bcfd-4d3f-acb7-dfa4e978ddd8">

> What is the actual difference (in percent)? 
Difference in percent is definitely lower. I extracted some pixels of the previous image an this is the result:
<img width="736" alt="image" src="https://github.com/photopea/photopea/assets/7047501/34d2418e-ffe1-47d2-b394-6184bcd6f81a">

So what you did is already a step forward in making the computation similar to what Photoshop does.

> Why do you think it is "correct" in Photoshop and not in Photopea? :D
You are right, I did not want to say Photoshop is correct.
But I'm trying to allow people of a community to not be force to use Photoshop to create PSD files, but in order to do that results should be identical.

Thanks again for your effort.
